modules/services/notes.js

```diff
@@ -0,0 +1,290 @@
+import _extend from 'lodash-es/extend';
+import _filter from 'lodash-es/filter';
+import _find from 'lodash-es/find';
+import _forEach from 'lodash-es/forEach';
+import _isEmpty from 'lodash-es/isEmpty';
+
+import osmAuth from 'osm-auth';
+
+import rbush from 'rbush';
+
+var _entityCache = {};
+
+import { dispatch as d3_dispatch } from 'd3-dispatch';
+import { xml as d3_xml } from 'd3-request';
+
+import { d3geoTile as d3_geoTile } from '../lib/d3.geo.tile';
+import { geoExtent } from '../geo';
+
+import {
+    osmNote,
+} from '../osm';
+
+import {
+    utilRebind,
+    utilIdleWorker
+} from '../util';
+
+var urlroot = 'https://api.openstreetmap.org',
```

Comment:
This shouldn't really be a hardcoded URL endpoint, right?

Reply:
thanks @mmd-osm !
It is hard coded in the other services including [osm.js](https://github.com/openstreetmap/iD/blob/7716532061d4e359621201ebf9b6aff709833f1a/modules/services/osm.js#L29). The remainder of the url includes the [notes portion](https://github.com/openstreetmap/iD/blob/1878397387b15c8c822c105fcfc8c74a37ab6185/modules/services/notes.js#L273) and then a [bounding box](https://github.com/openstreetmap/iD/blob/1878397387b15c8c822c105fcfc8c74a37ab6185/modules/services/notes.js#L225) once it is determined. What would you suggest?

Reply:
I was just wondering if you could run the Notes feature against a local database only (which is convenient for testing). As I have the osm.js part running in such a way, I'm assuming that there's some override mechanism in place. Looking very briefly at your implementation I couldn't find something similar for notes. Could be I simply missed it.

Reply:
@mmd-osm no I think you're right. There isn't anything in place to point to a local server. I will address this and check the incoming options to see if a different path is provided. Thoughts @bhousel ?

Reply:
Hmm yeah @mmd-osm and @thomas-hervey - it might make sense to refactor this notes service into `services/osm.js`, which already is setup to switch between dev and live servers.

For now, let's just test it how it is..